### PR TITLE
feat: add option to create remote repository before first push

### PR DESCRIPTION
**Pull request by morpheu**

### CODE REVIEW

**Review by crgisch — APPROVED** [review]
LGTM 👍

File: pkg/build/buildkit/build.go [hunk 1]
```diff
@@ -252,6 +268,29 @@ func (b *BuildKit) extractTsuruConfigsFromContainerImage(ctx context.Context, c
 	return callBuildKitToExtractTsuruConfigs(ctx, c, tmpDir, workingDir)
 }
 
+func (b *BuildKit) createRemoteRepository(ctx context.Context, r *pb.BuildRequest) error {
+	for _, v := range r.DestinationImages {
+		if provider, ok := b.opts.RemoteRepository[build.GetRegistry(v)]; ok {
+			err := provider.Auth(ctx)
+			if err != nil {
+				return err
```
Comment (wpjunior) [inline pkg/build/buildkit/build.go]: @morpheu I suggest to you simplify Provider interface in just one method: Ensure()

File: pkg/repository/repository.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"fmt"
+
+	"github.com/tsuru/deploy-agent/pkg/repository/fake"
+	"github.com/tsuru/deploy-agent/pkg/repository/oci"
+)
+
+type Repository interface {
+	Auth(ctx context.Context) error
+	Create(ctx context.Context, name string) error
+	Exists(ctx context.Context, name string) (bool, error)
```
Comment (wpjunior) [inline pkg/repository/repository.go]: @morpheu you could just have a single method: Ensure(ctx contet.Context, name string) error

**Review by wpjunior — CHANGES_REQUESTED** [review]

File: pkg/repository/repository.go (hunk 2 above)
Comment (morpheu) [reply pkg/repository/repository.go]: done :)

**Review by wpjunior — APPROVED** [review]

**Pull request merged**